### PR TITLE
rxtx: Support RX/TX

### PR DESCRIPTION
**Pull request by danak6jq**
Maintainer:  k6jq@comcast.net
Compile tested: OpenWRT Barrier Breaker, Chaos Calmer, trunk; aarch64 odroid C2, x64 virtualbox
Run tested: ar7xxx/9xxx Carambola2 & armv7 Raspberry Pi 2B; OpenWRT Barrier Breaker, Chaos Calmer,
tested with USB ACM with a Java application (Stoker II BBQ controller)

Description:

Add support for RX/TX

Signed-off-by: Dana H. Myers k6jq@comcast.net

### CODE REVIEW

**Comment by yousong** [comment]
This package should not be included in this repo unless we have proper java support in OpenWrt/LEDE.  I am closing this one for now.

**Comment by danak6jq** [comment]
This only depends on existing packages classpath and jamvam. Why did you close this? What am I missing?

**Comment by danak6jq** [comment]
On 12/16/2016 11:17 PM, Yousong Zhou wrote:
>
> This package should not be included in this repo unless we have proper java support in OpenWrt/LEDE. I am closing this one for now.
>
>

Please define "proper Java support" before closing this pull request.

Cheers,
Dana

**Comment by danak6jq** [comment]
On 12/16/2016 11:17 PM, Yousong Zhou wrote:
>
> Closed #3638 <https://github.com/openwrt/packages/pull/3638>.
>
>

Please explain this action.

Thank you,
Dana

**Comment by yousong** [comment]
> This only depends on existing packages classpath and jamvm. Why did you close this? What am I missing?

Sorry, I am not aware of this `jamvm` package providing Java support.

File: libs/rxtx/Makefile [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	$(CP) $(PKG_BUILD_DIR)/$(GNU_TARGET_NAME)-gnu/.libs/*.so \
+		$(1)/usr/lib/classpath/
+	$(CP) $(PKG_BUILD_DIR)/RXTXcomm.jar \
+		$(1)/usr/lib/classpath/
+endef
+
+define Build/Install
+#
+endef
+
```
Comment (yousong) [inline libs/rxtx/Makefile]: Remove the `PKG_INSTALL:=1` line above, then overriding `Build/Install` is also not necessary

File: libs/rxtx/Makefile [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+PKG_NAME:=rxtx
+PKG_VERSION:=2.2pre2
+PKG_RELEASE:=1
+PKG_LICENSE:=LGPL-2.1
+PKG_MAINTAINER:=Dana H. Myers <k6jq@comcast.net>
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).zip
+PKG_SOURCE_URL:=http://rxtx.qbang.org/pub/rxtx/
+PKG_MD5SUM:=
```
Comment (yousong) [inline libs/rxtx/Makefile]: Normally file checksum of static tarballs (other than those made from git/svn repos) should be provided and if this is not the case, a minor comment in the code may help.

File: libs/rxtx/Makefile [hunk 3]
```diff
@@ -0,0 +1,54 @@
+#
+# Copyright (C) 2006-2016 OpenWrt.org
```
Comment (yousong) [inline libs/rxtx/Makefile]: This copyright header is outdated and it is recommended that the original author of the content take the copyright unless they really have a signed contract with OpenWrt.org

**Comment by yousong** [comment]
Please also squash the 2 commits into a single one and provide a Signed-off-by line according to the `CONTRIBUTING.md` guide

**Comment by danak6jq** [comment]
Thank you - I was thinking OpenWRT had lost it's collective mind to refuse
a free offering of real value in a period of time that OpenWRT is under
administrative attack.

I'll review and incorporate the suggestions.

Thank you,
Dana

File: libs/rxtx/Makefile [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+define Package/rxtx
+  SECTION:=libs
+  CATEGORY:=Libraries
+  TITLE:=RX/TX
+  URL:=http://rxtx.qbang.org/wiki/index.php/Main_Page
+  DEPENDS:=+libpthread
+endef
+
+define Package/rxtx/Description
+	RX/TX Support for serial communications
```
Comment (yousong) [inline libs/rxtx/Makefile]: Actually it took me a while to find out that rxtx is a java library.  I think the `TITLE` and `Package/rxtx/description` (note the typo) can be more informative.

Excerpt taken from rxtx's FAQ page

```
RXTX is a Java library, using a native implementation (via JNI), providing serial and parallel communication for the Java Development Toolkit (JDK).
```

**Comment by danak6jq** [comment]
On 12/17/2016 12:22 AM, Yousong Zhou wrote:
>
> Please also squash the 2 commits into a single one and provide a Signed-off-by line according to the |CONTRIBUTING.md| guide
>
>

My initial pull request contains:

Signed-off-by: Dana H. Myers k6jq@comcast.net


Is this not adequate?

Thank you,
Dana

**Comment by hnyman** [comment]
> Is this not adequate?

You still need to add the signed-off-by line to the commit message itself (and there needs to be a proper commit message), as the commit message gets stored into the git log, unlike this PR discussion.

You also need to squash the five commits into one. You have the commits is a separate branch, so it is easy. Just use `git rebase -i HEAD~5` in the branch and squash the commits into one (and edit the commit message at the same time). Advice in https://github.com/openwrt/packages/blob/master/CONTRIBUTING.md

**Comment by danak6jq** [comment]
On 12/19/2016 1:39 PM, Hannu Nyman wrote:
>
>     Is this not adequate?
>
> You still need to add the signed-off-by line to the commit message itself (and there needs to be a proper commit message), as 
> the commit message gets stored into the git log, unlike this PR discussion.
>

D'oh. Of course. My apologies.

> You also need to squash the five commits into one. You have the commits is a separate branch, so it is easy. Just use |git 
> rebase -i HEAD~5| in the branch and squash the commits into one (and edit the commit message at the same time). Advice in 
> https://github.com/openwrt/packages/blob/master/CONTRIBUTING.md
>

Yup - thanks. Working on this now...

Cheers,
Dana

**Comment by danak6jq** [comment]
So of course I totally poked my own git-eye out attempting to do a rebase of the previous
branch, started fresh with command-line git tools that don't confuse and frighten as much
as github desktop.

I think I've done the right thing, and opened #3679 <https://github.com/openwrt/packages/pull/3679> and have my fingers crossed
all the bits are lined-up correctly this time.

Cheers,
Dana
P.S. New Year's resolution: actually learn how to use git

**Pull request closed without merging**